### PR TITLE
Add tests for `gen`

### PR DESCRIPTION
**Pull request by bsilver8192**
Split out from google/autocxx#876.

### CODE REVIEW

File: demo/src/main.rs [hunk 1]
```diff
@@ -21,6 +21,9 @@ include_cpp! {
     generate!("Goat")
 }
 
+#[link(name = "autocxx-demo")]
+extern "C" {}
+
```
Comment (adetaylor) [inline demo/src/main.rs:26]: Still can't do this, sorry.

Comment (adetaylor) [reply demo/src/main.rs:26]: Ping! @bsilver8192 do you plan to address this? Otherwise, I'll get around to it in the end.

Comment (bsilver8192) [reply demo/src/main.rs]: Sorry, forgot about this. Done

**Comment by adetaylor** [comment]
Great. Thanks!

**Pull request merged**